easybuild/easyblocks/n/neuron.py

```diff
@@ -60,85 +62,136 @@ def extra_options():
         extra_vars = {
             'paranrn': [True, "Enable support for distributed simulations.", CUSTOM],
         }
-        return ConfigureMake.extra_options(extra_vars)
+        return CMakeMake.extra_options(extra_vars)
 
     def configure_step(self):
         """Custom configuration procedure for NEURON."""
+        if LooseVersion(self.version) < LooseVersion('7.8.1'):
+            # enable support for distributed simulations if desired
+            if self.cfg['paranrn']:
+                self.cfg.update('configopts', '--with-paranrn')
+
+            # specify path to InterViews if it is available as a dependency
+            interviews_root = get_software_root('InterViews')
+            if interviews_root:
+                self.cfg.update('configopts', "--with-iv=%s" % interviews_root)
+            else:
+                self.cfg.update('configopts', "--without-iv")
+
+            # optionally enable support for Python as alternative interpreter
+            python_root = get_software_root('Python')
+            if python_root:
+                self.with_python = True
+                self.cfg.update('configopts', "--with-nrnpython=%s/bin/python" % python_root)
+
+            # determine host CPU type
+            cmd = "./config.guess"
+            (out, ec) = run_cmd(cmd, simple=False)
 
-        # enable support for distributed simulations if desired
-        if self.cfg['paranrn']:
-            self.cfg.update('configopts', '--with-paranrn')
+            self.hostcpu = out.split('\n')[0].split('-')[0]
+            self.log.debug("Determined host CPU type as %s" % self.hostcpu)
 
-        # specify path to InterViews if it is available as a dependency
-        interviews_root = get_software_root('InterViews')
-        if interviews_root:
-            self.cfg.update('configopts', "--with-iv=%s" % interviews_root)
+            # determine Python lib dir
+            self.pylibdir = det_pylibdir()
+
+            # complete configuration with configure_method of parent
+            ConfigureMake.configure_step(self)
         else:
-            self.cfg.update('configopts', "--without-iv")
+            # enable support for distributed simulations if desired
+            if self.cfg['paranrn']:
+                self.cfg.update('configopts', '-DNRN_ENABLE_MPI=ON')
+            else:
+                self.cfg.update('configopts', '-DNRN_ENABLE_MPI=OFF')
 
-        # optionally enable support for Python as alternative interpreter
-        python_root = get_software_root('Python')
-        if python_root:
-            self.with_python = True
-            self.cfg.update('configopts', "--with-nrnpython=%s/bin/python" % python_root)
+            # specify path to InterViews if it is available as a dependency
+            interviews_root = get_software_root('InterViews')
+            if interviews_root:
+                self.cfg.update('configopts', "-DIV_DIR=%s" % interviews_root)
```

Comment:
Strictly speaking one should also make sure -DNRN_ENABLE_INTERVIEWS=ON here in case they change the default.